### PR TITLE
Add quickstart vignette

### PR DESCRIPTION
**Pull request by aaronwolen**
## Features

- New [quickstart vignette](https://github.com/TileDB-Inc/tiledbsc/blob/aaronwolen/sc-20254/created-hosted-example-notebooks-for-tiledbsc/vignettes/quickstart.Rmd.orig) which provides new users with a high-level overview of the package
- New function `dataset_seurat_pbmc3k()` to download the pbmc 3k dataset from 10X and import as a `Seurat` object without requiring any extra dependencies. This dataset is used in the new vignette
- Updated bundled `Makefile` to add targets for generating pre-computed vignettes and performing common dev operations
- Added [`CONTRIBUTING.md`] to reference TileDB's CoC and document the `Makefile`

## Minor changes

- added a few missing `import` statements noted by R CMD check

## Vignette 

The new vignette covers the following topics:

- ingesting *Seurat* data into TileDB-backed SOMA Collection locally or on TileDB Cloud
- accessing various components of a dataset and reading into memory
- slicing by cell-/feature-identifiers
- querying w/ cell-/feature-level annotations
- iterative construction of a SOMA through a standard analysis
- loading data from SOMA into R as a *Seurat* or *Bioconductor* object

*Note: This vignette will also live on TileDB Cloud as a public notebook.*

## Details

Because this vignette makes use of an externally hosted dataset that must be downloaded first _and_ utilizes functions from *Seurat*, which is _not_ a tiledbsc dependency, it must be _precomputed_ (i.e., `vignettes/quickstart.Rmd` is created locally by knitting/rendering `vignettes/quickstart.Rmd.orig`). See [here](https://ropensci.org/blog/2019/12/08/precompute-vignettes/) for information about the approach.

For developers/contributors, this means `vignettes/quickstart.Rmd` must be regenerated after making edits to `vignettes/quickstart.Rmd.orig`. This process can be performed automatically by running `make vignettes`.

### CODE REVIEW

**Review by johnkerl — APPROVED** [review]
Awesome!!! :D 

This part from the PR description:

> Because this vignette makes use of an externally hosted dataset that must be downloaded first and utilizes functions from Seurat, which is not a tiledbsc dependency, it must be precomputed (i.e., vignettes/quickstart.Rmd is created locally by knitting/rendering vignettes/quickstart.Rmd.orig). See [here](https://ropensci.org/blog/2019/12/08/precompute-vignettes/) for information about the approach.
> 
> For developers/contributors, this means vignettes/quickstart.Rmd must be regenerated after making edits to vignettes/quickstart.Rmd.orig. This process can be performed automatically by running make vignettes

should be copied verbatim to the `Makefile` as a block comment.

File: vignettes/quickstart.Rmd [hunk 1]
```diff
@@ -0,0 +1,3445 @@
...
+  %\VignetteEncoding{UTF-8}
+---
+
+## Overview
+
+This vignette provides an overview of the tiledbsc R package which uses the open source [SOMA data model](https://github.com/single-cell-data/SOMA) to store single-cell data in a collection of TileDB arrays.
+
+With tiledbsc you can:
+
+- convert datas from Seurat and other popular single-cell formats into SOMA
```
Comment (johnkerl) [inline vignettes/quickstart.Rmd:16]: convert datas -> convert data

Comment (johnkerl) [reply vignettes/quickstart.Rmd:16]: into SOMA -> into SOMA format

File: vignettes/quickstart.Rmd [hunk 2]
```diff
@@ -0,0 +1,3445 @@
...
+- access SOMA data using any of TileDB's APIs
+- store your datasets on S3 (or other remote object stores) and slice directly without having to download anything first
+
+In this vignette you'll learn how to:
+
+- ingest *Seurat* data into TileDB-backed SOMA Collections
```
Comment (johnkerl) [inline vignettes/quickstart.Rmd:22]: i could do without italicizing Seurat (my $0.02)

File: vignettes/quickstart.Rmd [hunk 3]
```diff
@@ -0,0 +1,3445 @@
...
+- access various components of a dataset
+- slice by cell-/feature-identifiers
+- query using cell-/feature-level annotations
+- update a SOMA with analysis results
+- load data from SOMA into R as a *Seurat* or *Bioconductor* object
```
Comment (johnkerl) [inline vignettes/quickstart.Rmd:27]: ditto

File: vignettes/quickstart.Rmd [hunk 4]
```diff
@@ -0,0 +1,3445 @@
...
+
+## Overview
+
+This vignette provides an overview of the tiledbsc R package which uses the open source [SOMA data model](https://github.com/single-cell-data/SOMA) to store single-cell data in a collection of TileDB arrays.
+
+With tiledbsc you can:
+
+- convert datas from Seurat and other popular single-cell formats into SOMA
+- access SOMA data using any of TileDB's APIs
+- store your datasets on S3 (or other remote object stores) and slice directly without having to download anything first
```
Comment (johnkerl) [inline vignettes/quickstart.Rmd:18]: you should include TileDB Cloud in this list

**Review by eddelbuettel — APPROVED** [review]
Looks good, few small comments made.  Nothing major so approving.

Haven't built or run it yet.  Maybe later today.

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+* YES: you edit a roxygen comment in a `.R` file below `R/`.
+* NO: you edit an `.Rd` file below `man/`.
+
+### Prerequisites
+
+Before you make a substantial pull request, you should always file an issue and
+make sure someone from the team agrees that it’s a problem. If you’ve found a
+bug, create an associated issue and illustrate the bug with a minimal
+[reprex](https://www.tidyverse.org/help/#reprex).
```
[the remark this answers is not in the document]
Comment (aaronwolen) [reply .github/CONTRIBUTING.md:20]: I think I'll drop this block, it was boiler plate that I copied from osfr for the Makefile changes/docs.

File: NEWS.md [hunk 6]
```diff
@@ -11,6 +11,10 @@
   - `AssayMatrix$to_matrix()`
 
 * Members can now be removed from `TileDBGroup`s with `remove_member()`
+* New `vignette("quickstart")` which provides new users with a high-level overview of the package
+* New function `dataset_seurat_pbmc3k()` to download the pbmc 3k dataset from 10X and import as a `Seurat` object without requiring any extra dependencies. This dataset is used in the new vignette
+* Updated bundled `Makefile` to add targets for generating pre-computed vignettes and performing common dev operations
```
[the remark this answers is not in the document]
Comment (aaronwolen) [reply NEWS.md:16]: I don't think so... Is there anything in particular I should look out for?

[the remark this answers is not in the document]
Comment (eddelbuettel) [reply NEWS.md:16]: If `R CMD check --as-cran` is quiet, you are good.  If it isn't, then add 'GNU Make' to SystemRequirements.  Worth it for a richer `make`.

File: vignettes/quickstart.Rmd (hunk 2 above)
Comment (aaronwolen) [reply vignettes/quickstart.Rmd:22]: Done!

File: vignettes/quickstart.Rmd.orig [hunk 7]
```diff
@@ -0,0 +1,291 @@
+---
```
[the remark this answers is not in the document]
Comment (aaronwolen) [reply vignettes/quickstart.Rmd.orig:1]: Yes, `quickstart.Rmd.orig` contains the source from which `quickstart.Rmd` is rendered. Noted in the PR description that I'm following Jeroen's approach outlined [here](https://ropensci.org/blog/2019/12/08/precompute-vignettes/).

File: vignettes/quickstart.Rmd (hunk 1 above)
Comment (aaronwolen) [reply vignettes/quickstart.Rmd:16]: Thanks! I'm incorporating your suggestions into `quickstart.Rmd.orig`, which contains the vignette source material.

File: vignettes/quickstart.Rmd.orig (hunk 7 above)
[the remark this answers is not in the document]
Comment (eddelbuettel) [reply vignettes/quickstart.Rmd.orig:1]: That is extremely counter-intuitive to users of 40-year tools like `diff` and `patch` which create or use `.orig` with a totally different meaning.  If you like the workflow, can you you settle for a different naming scheme?  Maybe 'prelim-source-file' being quickstart.Rmd.in as in autoconf and friends?   Or .input or ... ?  Anyway.

**Review by johnkerl — COMMENTED** [review]
did i leave a review in pending? confused ...

File: vignettes/quickstart.Rmd (hunk 4 above)
Comment (aaronwolen) [reply vignettes/quickstart.Rmd:18]: I added a bullet indicating the vignette also covers how to store data on s3 and register w/ TileDB Cloud

File: vignettes/quickstart.Rmd [hunk 8]
```diff
@@ -0,0 +1,3445 @@
...
+## Reading AnnotationDataframe into memory from '/var/folders/rn/r3r2rnt56z98l344kf1tq5ph0000gn/T//RtmpEdCxNw/soco-pbmc3k/soma_RNA/obs'
+```
+
+```
+##                     orig.ident nCount_RNA nFeature_RNA
+## AAACATACAACCAC-1 SeuratProject       2421          781
+## AAACATTGAGCTAC-1 SeuratProject       4903         1352
+## AAACATTGATCAGC-1 SeuratProject       3149         1131
+## AAACCGTGCTTCCG-1 SeuratProject       2639          960
+## AAACCGTGTATGCG-1 SeuratProject        981          522
```
[the remark this answers is not in the document]
Comment (aaronwolen) [reply vignettes/quickstart.Rmd:361]: Oh good catch!

File: vignettes/quickstart.Rmd.orig (hunk 7 above)
[the remark this answers is not in the document]
Comment (aaronwolen) [reply vignettes/quickstart.Rmd.orig:1]: Good point. Would `.source` be clearer?

**Comment by aaronwolen** [comment]
Thanks @johnkerl and @eddelbuettel for the valuable feedback!

File: vignettes/quickstart.Rmd.source [hunk 9]
```diff
@@ -224,7 +224,7 @@ soma$X
 
 ### Add Dimensional Reduction Results
 
-We can use a similar workflow for dimensional reduction results, which are stored in either the `obsm` or `varm` slots of a SOMA.
+We can use a similar workflow for dimensional-reduction results, which are stored in either the `obsm` or `varm` slots of a SOMA.
```
Comment (johnkerl) [inline vignettes/quickstart.Rmd.source:227]: dimension-reduction results

**Pull request merged**